### DISCUSSION TITLE
ReaLearn excessive midi latency

### DISCUSSION THREAD

**Discussion by sekim6** [helgoboss/helgobox#1336] [General]
ReaLearn is adding significant latency when used to connect a CC outputting modulator (i.e. Melda MCCGenerator) and a mapped plugin parm (in this example Melda MBandpass low pass filter cutoff freq). The latency is very evident in retrigger scenarios. However, connecting MCC via Reaper's parm/CC mapper does not exhibit excessive latency in the same retrigger scenarios. Nor does adding a CC mapping layer (PIZ midi16CCRouter) in between MCC and Reaper's parm/CC mapper. Is the excessive latency introduced by ReaLearn considered "normal"?

**Comment by helgoboss** [comment]
I mentioned this in the thread. This is the number one reason why I don't consider ReaLearn as a real modulation solution at the moment: It doesn't yet do parameter modulation in real-time.

However, I will add this in the next release. Then you will be able to control FX params in real-time, but only if they are on the same track as ReaLearn and only FX params (it won't work with any of the other targets).

**Reply by sekim6** [reply]
Sounds good, looking forward to the update.

**Reply by helgoboss** [reply]
I finally scheduled this for 2.16.16: https://github.com/helgoboss/helgobox/issues/1365.

**Reply by helgoboss** [reply]
@sekim6 It's here: Release 2.16.16.

Tick the checkbox "Real-time" in the target "FX parameter: Set value". Also make sure to check the requirements: https://docs.helgoboss.org/realearn/targets/fx-parameter/set-value.html

**Reply by xavmatazz** [reply]
Does that mean we should always load a realearn instance on every track, for example to control track volume in real time? Why not, but it's more time consuming to map, and to have an overview of what is already controlled, with which controller.

**Reply by helgoboss** [reply]
No. I strongly recommend against it. You should only consider it if you really need super low latency FX parameter control.

What's your use case?

**Reply by xavmatazz** [reply]
Thank you for your so quick and personal answer... so honoured. Let me congratulate for all your soft!!! I put all my hopes in realearn and chose Reaper for live performances with a singer. I play the drums and keyboards, and I sometimes want adaptative control over sonic parameters (cutoff, delay send...). I used to play on Nord G2. It provides great macro control abilities but is not so fast I would like for example between 2 slots or when combining several continuous parameters. Traffic can cause real problems. I wish realearn and reaktor could help me to build something stronger than my beloved G2. I already noticed a problem when I tried this : a realearn instance in a midi track (midi maped to pads) controls another track with a bunch of rs5k slices. Every hit increases midi note (I used increment to achieve that, but discovered this latency problem, and also another problem... how can I reset midi note?). Finally I used the G2 to bypass those problems. But I still believe in your so powerful and elegant realearn abilities.
